instrumentation/aws-sdk/aws-sdk-2.2/library/src/testBedrockRuntime/java/io/opentelemetry/instrumentation/awssdk/v2_2/internal/Aws2BedrockRuntimeTest.java

```diff
@@ -414,6 +425,409 @@ void testConverseToolCallNoMessageContent() {
                             KeyValue.of("index", Value.of(0)))));
   }
 
+  @Test
+  void testConverseToolCallStreamNoMessageContent()
+      throws InterruptedException, ExecutionException {
+    BedrockRuntimeAsyncClientBuilder builder = BedrockRuntimeAsyncClient.builder();
+    AwsSdkTelemetry telemetry =
+        AwsSdkTelemetry.builder(testing.getOpenTelemetry())
+            .setGenaiCaptureMessageContent(false)
+            .build();
+    builder.overrideConfiguration(
+        ClientOverrideConfiguration.builder()
+            .addExecutionInterceptor(telemetry.newExecutionInterceptor())
+            .build());
+    configureClient(builder);
+    BedrockRuntimeAsyncClient client = telemetry.wrapBedrockRuntimeClient(builder.build());
+
+    String modelId = "amazon.nova-micro-v1:0";
+    List<Message> messages = new ArrayList<>();
+    messages.add(
+        Message.builder()
+            .role(ConversationRole.USER)
+            .content(
+                ContentBlock.fromText("What is the weather in Seattle and San Francisco today?"))
+            .build());
+
+    StringBuilder responseChunksText = new StringBuilder();
+    List<ToolUseBlock.Builder> responseChunksTools = new ArrayList<>();
+    StringBuilder currentToolArgs = new StringBuilder();
+
+    ConverseStreamResponseHandler responseHandler =
+        ConverseStreamResponseHandler.builder()
+            .subscriber(
+                ConverseStreamResponseHandler.Visitor.builder()
+                    .onContentBlockStart(
+                        chunk -> {
+                          if (!responseChunksTools.isEmpty()) {
+                            JsonNode node = JsonNode.parser().parse(currentToolArgs.toString());
+                            currentToolArgs.setLength(0);
+                            Document document =
+                                node.visit(
+                                    new software.amazon.awssdk.protocols.json.internal.unmarshall
+                                        .document.DocumentUnmarshaller());
```

Comment:
as far as I can tell don't need the fully qualified name as it is already imported

Reply:
Oops, thanks